clcache.py

```diff
@@ -781,15 +788,10 @@ def expandBasedirPlaceholder(path):
 
 def collapseBasedirToPlaceholder(path):
     baseDir = normalizeBaseDir(os.environ.get('CLCACHE_BASEDIR'))
-    if baseDir is None:
-        return path
-    else:
-        assert path == os.path.normcase(path)
-        assert baseDir == os.path.normcase(baseDir)
-        if path.startswith(baseDir):
-            return path.replace(baseDir, BASEDIR_REPLACEMENT, 1)
-        else:
-            return path
+    if baseDir:
+        replacement = re.compile(re.escape(baseDir), re.IGNORECASE)
+        return replacement.sub(BASEDIR_REPLACEMENT, path)
+    return path
```

Comment:
Any reason you dropped the `assert` statements? They seem to express that the function expects that both the given `path` as well as the `baseDir` are normalized, i.e. have not only the same upper/lowercase but also forward slashes converted to backslashes.